src/key/helper.js

```diff
@@ -146,43 +146,58 @@ export async function getPreferredHashAlgo(key, keyPacket, date = new Date(), us
-    'aead': config.preferredAEADAlgorithm,
-    'compression': config.preferredCompressionAlgorithm
-  }[type];
-  const prefPropertyName = {
-    'symmetric': 'preferredSymmetricAlgorithms',
-    'aead': 'preferredAEADAlgorithms',
-    'compression': 'preferredCompressionAlgorithms'
-  }[type];
+export async function getPreferredCompressionAlgo(keys = [], date = new Date(), userIDs = [], config = defaultConfig) {
+  const defaultAlgo = enums.compression.uncompressed;
+  const preferredSenderAlgo = config.preferredCompressionAlgorithm;
 
   // if preferredSenderAlgo appears in the prefs of all recipients, we pick it
   // otherwise we use the default algo
   // if no keys are available, preferredSenderAlgo is returned
   const senderAlgoSupport = await Promise.all(keys.map(async function(key, i) {
-    const primaryUser = await key.getPrimaryUser(date, userIDs[i], config);
-    const recipientPrefs = primaryUser.selfCertification[prefPropertyName];
+    const selfCertification = await key.getPrimarySelfSignature(date, userIDs[i], config);
+    const recipientPrefs = selfCertification.preferredCompressionAlgorithms;
     return !!recipientPrefs && recipientPrefs.indexOf(preferredSenderAlgo) >= 0;
   }));
   return senderAlgoSupport.every(Boolean) ? preferredSenderAlgo : defaultAlgo;
 }
 
+/**
+ * Returns the preferred symmetric and AEAD algorithm (if any) for a set of keys
+ * @param {Array<Key>} [keys] - Set of keys
+ * @param {Date} [date] - Use the given date for verification instead of the current time
+ * @param {Array} [userIDs] - User IDs
+ * @param {Object} [config] - Full configuration, defaults to openpgp.config
+ * @returns {Promise<[module:enums.symmetric,module:enums.aead|0]>} Array containing the preferred symmetric algorithm, and the preferred AEAD algorithm, or 0 if CFB is preferred
+ * @async
+ */
+export async function getPreferredCipherSuite(keys = [], date = new Date(), userIDs = [], config = defaultConfig) {
+  const selfSigs = await Promise.all(keys.map((key, i) => key.getPrimarySelfSignature(date, userIDs[i], config)));
+  if (config.aeadProtect && selfSigs.every(selfSig => selfSig.features[0] & enums.features.seipdv2)) {
+    const defaultCipherSuite = [enums.symmetric.aes128, enums.aead.ocb];
+    const desiredCipherSuite = [config.preferredSymmetricAlgorithm, config.preferredAEADAlgorithm];
+    return selfSigs.every(selfSig => selfSig.preferredCipherSuites && selfSig.preferredCipherSuites.some(cipherSuite =>
+      cipherSuite[0] === desiredCipherSuite[0] && cipherSuite[1] === desiredCipherSuite[1])) ?
+      desiredCipherSuite :
+      defaultCipherSuite;
+  }
+  const defaultSymAlgo = enums.symmetric.aes128;
+  const desiredSymAlgo = config.preferredSymmetricAlgorithm;
+  return [
+    selfSigs.every(selfSig => selfSig.preferredSymmetricAlgorithms && selfSig.preferredSymmetricAlgorithms.includes(desiredSymAlgo)) ?
+      desiredSymAlgo :
+      defaultSymAlgo,
+    0
```

Comment:
Not a fan of this, I had to go and check whether `enums.aead.cfb === 0` since it got me confused; can we set `null` instead? Or just return `[<symAlgo>]`.
Based on the couple of calls I've seen, it's not clear to me whether we actually need an array -- can we can get away with `{ cipherAlgo, aeadAlgo }` ?

Reply:
:+1: I've changed it to `{ symmetricAlgo, aeadAlgo }`, with `undefined` meaning CFB.